src/shell.js

```diff
@@ -135,8 +135,10 @@ if (Module['ENVIRONMENT']) {
 // 2) We could be the application main() thread proxied to worker. (with Emscripten -sPROXY_TO_WORKER) (ENVIRONMENT_IS_WORKER == true, ENVIRONMENT_IS_PTHREAD == false)
 // 3) We could be an application pthread running in a worker. (ENVIRONMENT_IS_WORKER == true and ENVIRONMENT_IS_PTHREAD == true)
 
-// ENVIRONMENT_IS_PTHREAD=true will have been preset in worker.js. Make it false in the main runtime thread.
-var ENVIRONMENT_IS_PTHREAD = Module['ENVIRONMENT_IS_PTHREAD'] || false;
+// The way we signal to a worker that it is hosting a pthread is to append
+// ?phread=1 to the URL which means `self.location.search` will be set in this
+// case.
+var ENVIRONMENT_IS_PTHREAD = ENVIRONMENT_IS_WORKER && self.location.search;
```

Comment:
I think checking for `self.location.search` might be too greedy. For example, it may already be set if someone does this:
```js
var module = await Module({
  locateFile: (fileName, scriptDirectory) => `${scriptDirectory}${fileName}?v=123`,
});
```

Reply:
True, we should pick some unique query string I guess

Reply:
Done